### PR TITLE
docs: Add a speech generation sample

### PR DESCRIPTION
**Pull request by kdestin**
# Description

This pull request adds a sample for speech generation, and includes that snippet into the SDK's readme (`pwsh eng/scripts/Update-Snippets.ps1`).

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/openai/Azure.AI.OpenAI/tests/Samples/SpeechGeneration.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            Voice = AudioSpeechVoice.Alloy,
+            ResponseFormat = AudioSpeechOutputFormat.Mp3,
+            Speed = 0.8f
+        };
+
+        Response<BinaryData> response = await client.GetAudioSpeechAsync(speechOptions);
+
+        File.WriteAllBytes("myAudioFile.mp3", response.Value.ToArray());
+        #endregion
+
```
Comment (ninghu) [inline sdk/openai/Azure.AI.OpenAI/tests/Samples/SpeechGeneration.cs:36]: Please remove the blank line.

File: sdk/openai/Azure.AI.OpenAI/README.md [hunk 2]
```diff
@@ -694,6 +694,25 @@ Console.WriteLine($"Translation ({translation.Duration.Value.TotalSeconds}s):");
 Console.WriteLine(translation.Text);
 ```
 
+
+### Generate Speech with Text-to-Speech (TTS) models
+
+```C# Snippet:SpeechGeneration
+string text = "Hello World";
```
Comment (ninghu) [inline sdk/openai/Azure.AI.OpenAI/README.md:701]: I would remove this line of code if we want the code to look more concise.

File: sdk/openai/Azure.AI.OpenAI/tests/Samples/SpeechGeneration.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+using Azure.Identity;
+using NUnit.Framework;
+
+namespace Azure.AI.OpenAI.Tests.Samples;
+
+public partial class SpeechGeneration
+{
+    [Test]
+    [Ignore("Only verifying that the sample builds")]
+    public async Task TranscribeAudio()
```
Comment (ninghu) [inline sdk/openai/Azure.AI.OpenAI/tests/Samples/SpeechGeneration.cs:16]: Update test method name

File: sdk/openai/Azure.AI.OpenAI/README.md [hunk 4]
```diff
@@ -694,6 +694,25 @@ Console.WriteLine($"Translation ({translation.Duration.Value.TotalSeconds}s):");
 Console.WriteLine(translation.Text);
 ```
 
+
```
Comment (ninghu) [inline sdk/openai/Azure.AI.OpenAI/README.md:697]: There is only one blank line b/w samples. Please remove the additional line.

Comment (kdestin) [reply sdk/openai/Azure.AI.OpenAI/README.md]: Resolve by https://github.com/ninghu/azure-sdk-for-net/pull/11/commits/de73afce474165f9b31c22fb85dee73ec798bb03

File: sdk/openai/Azure.AI.OpenAI/tests/Samples/SpeechGeneration.cs (hunk 1 above)
Comment (kdestin) [reply sdk/openai/Azure.AI.OpenAI/tests/Samples/SpeechGeneration.cs]: Resolve by https://github.com/ninghu/azure-sdk-for-net/pull/11/commits/de73afce474165f9b31c22fb85dee73ec798bb03

File: sdk/openai/Azure.AI.OpenAI/tests/Samples/SpeechGeneration.cs (hunk 3 above)
Comment (kdestin) [reply sdk/openai/Azure.AI.OpenAI/tests/Samples/SpeechGeneration.cs]: Resolved by https://github.com/ninghu/azure-sdk-for-net/pull/11/commits/07347328bfaac7fb3dbc92bcf5b93145eb0e7a04

File: sdk/openai/Azure.AI.OpenAI/README.md (hunk 2 above)
Comment (kdestin) [reply sdk/openai/Azure.AI.OpenAI/README.md:700]: Resolved by https://github.com/ninghu/azure-sdk-for-net/pull/11/commits/1ccebca8e08f53c52cfe952b5674f4272be5f705

Comment (ninghu) [reply sdk/openai/Azure.AI.OpenAI/README.md:700]: I should have been more clear. What I meant was you can directly set "Hello World" to speechOptions.Input

Comment (kdestin) [reply sdk/openai/Azure.AI.OpenAI/README.md:700]: Resolved by https://github.com/ninghu/azure-sdk-for-net/pull/11/commits/97df7df2bcd0ed8913c577627cab60952e3bfc23

File: sdk/openai/Azure.AI.OpenAI/tests/Samples/SpeechGeneration.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            Input = text,
+            DeploymentName = "my-tts-deployment", // tts-1 as model name for non-Azure OpenAI
+            Voice = AudioSpeechVoice.Alloy,
+            ResponseFormat = AudioSpeechOutputFormat.Mp3,
+            Speed = 0.8f
+        };
+
+        Response<BinaryData> response = await client.GetAudioSpeechAsync(speechOptions);
+
```
[the remark this answers is not in the document]
Comment (kdestin) [reply sdk/openai/Azure.AI.OpenAI/tests/Samples/SpeechGeneration.cs]: Resolved by https://github.com/ninghu/azure-sdk-for-net/pull/11/commits/9f64c572635db6bd4e306950e60c8e722c70744a

**Review by ninghu — APPROVED** [review]

**Pull request merged**